### PR TITLE
Front End from Nava next.js template

### LINKED ISSUE #110: [Task]: Front-end Repo Setup
### Summary

Set up the front-end portion of the repository with the language decided in #96  and the set of code quality tools selected in #102  

### User Stories

- As an **open source contributor**, I want to:
  - be able to reference documentation explaining how to use the developer tools, so that I don't have to learn how to use these tools on my own in order to contribute to the project
  - have full test coverage for the codebase, so that I know when I've introduce code that changes or breaks existing behavior
  - have code formatting and standards enforced with automated tooling, so that I don't have to learn and check that my code adheres to those standards manually
- As a **maintainer of the project**, I want to:
  - have code quality checks run automatically on each push, so that formatting, linting, or security issues are caught before being deployed to production
  - ensure that new contributions meet certain thresholds for test coverage, so that code contributions from internal and external developers also include tests which validate the code's behavior

### Acceptance Criteria

- [x] A `frontend/` sub-directory has been created with a `README` that includes instructions for how to install and run the code in this sub-directory
- [x] At least 2 dependencies have been added to the project, and the dependency manager selected in #102 automatically checks for dependency conflicts and offers suggestions for how to reconcile those conflicts:
- [x] A dummy test has been created (i.e. `assert True`) that users can execute to verify they've installed the project correctly
- [x] The following checks are run before every commit:
  - [x] Auto-formatting
  - [x] Linting (basic)
  - [x] Type checks (optional)
  - ~[ ] Commit formatting (preferably conventional commit)~
- [x] The following checks are run with each push to the remote GitHub repo, and these checks are run against all minor versions of the backend language we plan to support:
  - [x] Auto-formatting
  - [x] Linting (complete)
  - [x] Type checks
  - [ ] ~Dependency vulnerability checks~
  - [ ] ~Open source license checks~
  - [x] Unit tests
  - [x] Test coverage
- [x] A PR is blocked from merging if any of the checks above fail
- [ ] At least 2 people who did not contribute to this PR have installed and run the dummy test locally

### Proposed Testing Strategy

1. **Unit tests:** Create a `test_sample` test that simply asserts `true` or asserts the version number
2. **Manual Tests:** Someone who did not contribute to the PR checks out the PR locally, follows the instructions in the README to install the code, run the tests and the code quality checks locally, and all checks/tests pass.

### PR DESCRIPTION
**Pull request by SammySteiner**
## Summary
Fixes #110 

### Time to review: 30 minutes

- confirm you can install and run the application locally
- confirm you can run the storybook locally
- confirm you can run the application on docker
- confirm you can run the storybook on docker 
- confirm you can run the tests

### CODE REVIEW

File: documentation/frontend/0000-use-markdown-architectural-decision-records.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Use Markdown Architectural Decision Records
```
Comment (lucasmbrown-usds) [inline documentation/frontend/0000-use-markdown-architectural-decision-records.md:1]: Should all these ADRs be moved into our ADR directory and merged with our ADRs in terms of numbering, etc?

File: .grype.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  - fix-state: not-fixed
+  - fix-state: wont-fix
+  - fix-state: unknown
+
+  # glob-parent before 5.1.2 vulnerable to Regular Expression Denial of Service in enclosure regex
+  # https://github.com/advisories/GHSA-ww39-953v-wcq6
+  # High severity
+  # Ignoring since this is only a dependency of dev tools: storybook (for storybook docs site),
+  # eslint (for Linting in CI), and sass (for compiling CSS during CI build phase)
+  - vulnerability: GHSA-ww39-953v-wcq6
```
Comment (lucasmbrown-usds) [inline .grype.yml:25]: Are these vulnerabilities still live?

**Review by lucasmbrown-usds — CHANGES_REQUESTED** [review]
Overall looks really good! left a couple comments, mostly I think the ADR comment needs to be addressed. 

Lots of fun bells and whistles with this one!

File: frontend/tests/components/Header.test.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import Header from "src/components/Header";
+
+describe("Header", () => {
```
Comment (lucasmbrown-usds) [inline frontend/tests/components/Header.test.tsx:6]: we love to see it!

File: frontend/tests/components/Layout.test.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        <h1>child</h1>
+      </Layout>
+    );
+
+    const header = screen.getByRole("heading", { name: /child/i, level: 1 });
+
+    expect(header).toBeInTheDocument();
+  });
+
+  it("passes accessibility scan", async () => {
```
Comment (lucasmbrown-usds) [inline frontend/tests/components/Layout.test.tsx:19]: nice

File: frontend/tests/jest.setup.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+require("@testing-library/jest-dom");
+const { toHaveNoViolations } = require("jest-axe");
+
+expect.extend(toHaveNoViolations);
```
Comment (lucasmbrown-usds) [inline frontend/tests/jest.setup.js:4]: very nice

File: .grype.yml (hunk 2 above)
Comment (SammySteiner) [reply .grype.yml]: Good call, I just double checked and it seems like they are no longer in the project.

Comment (SammySteiner) [reply .grype.yml]: Also, noting that when we add infrastructure via terraform, we'll have a github workflow that incorporates the grype scanning in the ci/cd pipeline.

File: documentation/frontend/0000-use-markdown-architectural-decision-records.md (hunk 1 above)
Comment (SammySteiner) [reply documentation/frontend/0000-use-markdown-architectural-decision-records.md:1]: Sure, I'll move those over.

File: .grype.yml (hunk 2 above)
Comment (daphnegold) [reply .grype.yml]: TIL anchore scan *is* grype

**Review by daphnegold — APPROVED** [review]
Amazing! :shipit: 🚀 A couple comments on small things like broken links

File: docker-compose.yml [hunk 6]
```diff
@@ -29,7 +29,37 @@ services:
...
+  nextjs:
+    container_name: next-dev
+    build:
+      context: ./frontend
+      target: dev
+    env_file:
+      # Add your non-secret environment variables to this file:
+      - ./frontend/.env.development
+      # If you have secrets, add them to this file and uncomment this line:
+      # - ./app/.env.local
```
Comment (daphnegold) [inline docker-compose.yml:41]:
```suggestion
      # - ./frontend/.env.local
```

File: documentation/frontend/internationalization.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
+# Internationalization (i18n)
+
+- [I18next](https://www.i18next.com/) is used for internationalization.
+- Next.js's [internationalized routing](https://nextjs.org/docs/advanced-features/i18n-routing) feature is enabled. Toggling between languages is done by changing the URL's path prefix (e.g. `/about` ➡️ `/es/about`).
+- Configuration for the i18n routing and i18next libraries are located in [`next-i18next.config.js`](../app/next-i18next.config.js).
```
Comment (daphnegold) [inline documentation/frontend/internationalization.md:5]: This local file link needs to be updated, it is broken

File: documentation/frontend/internationalization.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+   ```
+
+Refer to the [i18next](https://www.i18next.com/) and [react-i18next](https://react.i18next.com/) documentation for more usage docs.
+
+## Add a new language
+
+1. Edit `next-i18next.config.js` and add the language to `locales`, using the BCP47 language tag (e.g. `en` or `es`).
+1. Add a language folder, using the same BCP47 language tag: `mkdir -p public/locales/<lang>`
+1. Add a language file: `touch public/locales/<lang>/common.json` and add the translated content. The JSON structure should be the same across languages. However, non-default languages can omit keys, in which case the default language will be used as a fallback.
+1. Optionally, add a label for the language to the `locales` object in [`.storybook/preview.js`](../app/.storybook/preview.js)
```
Comment (daphnegold) [inline documentation/frontend/internationalization.md:56]: This file path needs to be updated, broken too

File: docker-compose.yml (hunk 6 above)
Comment (SammySteiner) [reply docker-compose.yml:41]: good catch! Thank you

**Pull request merged**